### ISSUE TITLE
When set sml/theme to 'dark,  unexpected disable `delete-selection-mode` 

### ISSUE THREAD

**Issue by zw963** [Malabarba/smart-mode-line#179]
I often use `delete-selectioin-mode` to speed up edit experience.

Which can replace a marked(highlighted) regin with input/paste directly.

so, I enable this mode with:
```cl
(delete-selection-mode 1)
```
But, when setting ``(setq sml/theme 'dark)``, it change this mode again.

I think this should not a expected behavior, confuse me a lot.

Thanks.

my config:

```cl
(require 'smart-mode-line)
(setq
 sml/theme 'dark
 sml/no-confirm-load-theme t
)
(sml/setup)
```

**Comment by Malabarba** [comment]
Do you have delete-selection-mode in your custom-settings?

**Comment by zw963** [comment]
Yes, I settings `(delete-selection-mode 1)` before settings smart-mode-line.

**Comment by zw963** [comment]
I have no try to settings `(delete-selection-mode 1)` after smart-mode-line though, because my all emacs
self-relative settings is set early than third party package settings.

**Comment by zw963** [comment]
just for clarify, `delete-selection-mode` is included in emacs, so i settings it early.

**Comment by Malabarba** [comment]
Sorry, I wasn't clear. That's not what I meant. Inside your init file, there should be a section that looks like this:

    (custom-set-variables
     ;; custom-set-variables was added by Custom.
     ;; If you edit it by hand, you could mess it up, so be careful.
     ;; Your init file should contain only one such instance.
     ;; If there is more than one, they won't work right.
     [... Lots of lines in here ...]
     )

Do you see `delete-selection-mode` somewhere in that form?

**Comment by zw963** [comment]
All my dotfiles is write by myself maually.  I could ensure I could not add any
`(custom-set-variables ... )` section in my dotfiles.

and I does not use ELPA too. (only for analysis)

some times, `(custom-set-variables ...)` is inserted into my ~/.emacs by some snippets code,
I will remove it when I found it.

**Comment by zw963** [comment]
Another clue:
my dotfiles is huge, when I test it, I just leave necessory package.

For this issue, I only have following config:
```cl
(delete-selection-mode t)
(require 'smart-mode-line)
(setq
 sml/theme 'dark
 sml/no-confirm-load-theme t
 )
(sml/setup)

(error "Done")
```
And, when i start emacs,  `delete-selection-mode` variable is nil

**Comment by Malabarba** [comment]
Still, it sounds like a custom-variables issue. I cannot imagine how sml itself would be turning off this mode for you.

Try the following: 

1. Start Emacs normally (with all your configs as and everything).
2. Issue `M-x customize-variable` and type `delete-selection-mode`.
3. You'll get a customization window. Click the “Toggle button” to set it to “on”, then type `C-x C-s` to save it.

**Comment by zw963** [comment]
> Start Emacs normally (with all your configs as and everything).
> Issue M-x customize-variable and type delete-selection-mode.
> You'll get a customization window. Click the “Toggle button” to set it to “on”, then type C-x C-s to save it.

This is work.  no doubt.

because emacs add some lisp code at the bottom of my ~/.emacs.

```cl
(defun relative-load (file-relative-path)
  (let ((absolute-path (expand-file-name
                        file-relative-path
                        (file-name-directory (or load-file-name buffer-file-name)))))
    (when (file-exists-p absolute-path) (load absolute-path))))

;; this is all my config files entry.
(relative-load "Dropbox/common/.emacs.d/init.el")

;;--------------------- following is add by `customize-varible`. ------------

(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(delete-selection-mode t)
 '(session-use-package t nil (session))
 '(tabbar-separator (quote (0.5))))
(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 )
```

In my case, i setting `(delete-selection-mode t)` before sml.
and sml disable my delete-selection-mode unexpectedly.

I just wonder the cause sml disable delete-selection-mode.

Could you please reproduce this ? my emacs version is 24.5.1.

keep a empty `$HOME/.emacs` file, and setting as following should reproduce:

```cl
;; add sml package to load-path.

(require 'smart-mode-line)
(setq
 sml/theme 'dark
 sml/no-confirm-load-theme t
)
(sml/setup)
```

**Comment by Malabarba** [comment]
Sorry, but I can't reproduce that. And there isn't a single reference to `delete-selection-mode` in the sml code files.

All I can say is that it must be something else in your configs. :-/

**Comment by zw963** [comment]
a completely process.

### step 1, edit ~/.emacs to as following:
```cl
(delete-selection-mode 1)
(add-to-list 'load-path
             "/home/zw963/Dropbox/common/.emacs.d/my-customized-settings/plugins/smart-mode-line")
(add-to-list 'load-path
             "/home/zw963/Dropbox/common/.emacs.d/my-customized-settings/plugins/smart-mode-line/themes")

(require 'smart-mode-line)
(setq sml/no-confirm-load-theme t)
(sml/setup)
```

### step 2: copy Github newst version (9dd588d) to load path (with dependency)
```sh
 ╭─ 19:40  zw963 ⮀ ~/Dropbox/common/.emacs.d/my-customized-settings/plugins/smart-mode-line ⮀ ⭠ (9dd588d) refactor *+% ➦ ruby-2.2.3
 ╰─ $ find -name '*.el' |xargs -n1 md5sum
075110d335eaacdc87ab694e90bffa45  ./smart-mode-line-dark-theme.el
00397cf05d85e6f5e0c8f29d48cf2253  ./smart-mode-line-light-theme.el
85bd551fa8d4473141c6bef29a146ab6  ./smart-mode-line-respectful-theme.el
3d61ac2f2adf658aabbe7dd8e2dbd56c  ./smart-mode-line.el
2577e351ac62d1c5bee921d6c7ccfd2d  ./themes/smart-mode-line-light-powerline-theme.el
a42dad2f090f2d5c6fd95687f1e6fb98  ./themes/smart-mode-line-powerline-theme.el
20ad3c3c29d3ad791c35547d4ff75aad  ./rich-minority.el
```

### step 3: start emacs with -Q

### step 4: open ~/.emacs, run eval-buffer

### step5: C-h v, delete-selection-mode is nil

**Comment by Malabarba** [comment]
I've followed those steps on Emacs 25 and 24.5.  In both cases `delete-selection-mode` was `t`.

**Comment by zw963** [comment]
Do you verify the md5 of lisp files?

**Comment by Malabarba** [comment]
Yes
